src/producingChangeset.html

```diff
@@ -0,0 +1,485 @@
+Reviewed-by: iag
+$ hg commit -l ../message
+$ hg toutgoing
+[.]
+comparing with http://hg.openjdk.java.net/sandbox/box
+searching for changes
+changeset:   23:fb12953f3a35
+tag:         tip
+user:        iris
+date:        Wed Dec 12 21:05:59 2007 -0800
+summary:     1111111: Missing Duke gif
+</pre></blockquote>
+
+<h2><a name="merge">Merging</a></h2>
+
+<p> It is often necessary to merge local changes with those made in the parent
+repositories.  The first step in a merge process is to retrieve (or pull) the
+collection of changesets which have been pushed since the last merge or initial
+clone.  If there if there are merge conflicts, then they must be resolved.
+<a href="http://hgbook.red-bean.com/hgbookch3.html#x7-530003">Chapter 3</a> of
+the Mercurial book contains detailed information on the merging process. </p>
+
+<p> There are two basic ways to update the working set files in the
+repositories: </p>
+
+<dl>
+<dt>Option 1: <code><i><b>hg</b></i> pull</code></dt>
+<dd>One way to merge the parent repository with the working set of files is to
+use <code><i><b>hg</b></i> pull</code> all by itself.
+This option allows merging off-line or at a later time. 
+
+<blockquote><pre>
+$ <code><i><b>hg</b></i></code> pull
+[.]
+pulling from http://hg.openjdk.java.net/sandbox/box
+searching for changes
+no changes found
+</pre></blockquote>
+
+<p> In Mercurial, pulling changesets will not update or merge into the working
+set of files.  To update the clone,
+run <code><i><b>hg</b></i> update</code>.  If the update
+reports conflicts,
+run <code><i><b>hg</b></i> merge</code> to resolve them. </p>
+</dd>
+
+<dt>Option 2: <code><i><b>hg</b></i> fetch</code></dt>
+<dd>Alternatively,
+use <code><i><b>hg</b></i> fetch</code> to pull the changes, update
+the working set files, and create simple merge changesets as necessary.
+The fetch extension is distributed with Mercurial but needs to be enabled.
+Edit the <code>.hgrc</code> to include the following entries:
+
+<blockquote><pre>
+[extensions]
+fetch=
+</pre></blockquote>
+
+<p> Once the fetch extension has been enabled,
+    <code><i><b>hg</b></i> fetch</code> may be
+invoked as follows: </p>
+
+<blockquote><pre>
+$ <code><i><b>hg</b></i></code> fetch
+[.]
+pulling from http://hg.openjdk.java.net/sandbox/box
+searching for changes
+no changes found
+</pre></blockquote></dd>
+
+</dl>
+
+<blockquote><hr />
+
+<p> Actual file merging will be done with the selected Mercurial merging tool
+see <a href="http://www.selenic.com/mercurial/wiki/index.cgi/MergeProgram">
+MergeProgram</a> for the details on how to define the selected merge tool in
+<code> ~/.hgrc</code>. </p>
+
+<hr /></blockquote>
+
+<h2><a name="push">Pushing</a></h2>
+
+<p> In order to push changesets into the parent repository, some additional
+configuration is required.  The following sections describe the operations that
+will be performed by users with push access. </p>
+
+<h3><a name="sshGen">Generating an SSH Key</a></h3>
```

Comment:
I hope this or similar anchor stays around long term. I often use it when answering user questions.

Reply:
The ATX headers that is recommended in other OpenJDK documents that use markdown automatically creates links to all headers. The default is to use the text from the header so the header above would be written as

>    ### Generating an SSH Key

which in turn would generate this html

<h3 id="generating-an-ssh-key">Generating an SSH Key</h3>

So all headers will have links but by default they won't be the same links as they used to be. I would prefer if we could stick with the defaults if possible.

Reply:
I agree.  We should use default links whenever possible.  My only concern is that the new links are stable.